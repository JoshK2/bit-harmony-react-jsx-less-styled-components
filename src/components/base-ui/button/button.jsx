import styled from 'styled-components';

export const Button = styled.button`
  font-size: 1em;
  padding: 0.25em 1em;
  border-radius: 3px;
  cursor: pointer;
  color: #44c57e;
  border: 2px solid #44c57e;
  outline: none;
  &:active {
    background-color: white;
  }
`;
