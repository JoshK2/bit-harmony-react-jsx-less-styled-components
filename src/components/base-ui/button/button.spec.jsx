import React from 'react';
import testRenderer from 'react-test-renderer';
import { render } from '@testing-library/react';
import { BasicButton } from './button.composition';

describe('Button component', () => {
  it('renders correctly', () => {
    testRenderer.create(<BasicButton />);
  });
  it('snapshot test', () => {
    const component = testRenderer.create(<BasicButton />);
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
  it('renders correctly with value', () => {
    const { getByText } = render(<BasicButton />);
    expect(getByText(/Click/)).toBeTruthy();
  });
});
