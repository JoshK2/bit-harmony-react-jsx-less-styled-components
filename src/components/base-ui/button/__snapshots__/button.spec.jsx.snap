// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button component snapshot test 1`] = `
<button
  className="sc-bdfBwQ kvbBrN"
>
  Click
</button>
`;
